app/models/category.rb

```diff
@@ -7,6 +7,7 @@ class Category < ApplicationRecord
   has_many :sub_categories, as: :categorizable, dependent: :destroy, class_name: 'Category'
 
   scope :main_category, -> { where(categorizable_type: 'User') }
+  scope :categories_have_budget, -> { where.not(amount: 0) }
```

Comment:
@marderer what do you think about `with_amount` instead of `categories_have_budget`. we already in `category` model, why do we need to make a note about it one more?